lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt

```diff
@@ -60,60 +61,64 @@ internal class NadelExecutionPlanFactory(
         rootField: ExecutableNormalizedField,
         serviceHydrationDetails: ServiceExecutionHydrationDetails?,
     ): NadelExecutionPlan {
-        val executionSteps = mutableListOf<AnyNadelExecutionPlanStep>()
```

Comment:
We used to do `.groupBy { it.field }` at the end but that is pretty wasteful.

So here we construct the `Map` directly.